### PR TITLE
Audio breakout rooms

### PR DESCRIPTION
**Pull request by Johnny850807**
## Why need this change? / Root cause: 
- Add audio-breakout rooms feature
## Changes made:
- 
The audio-breakout module is added to provide the Audio Breakout Rooms feature. This commit also introduces JDA event GenericCommandInteractionEvent's extension methods to handle the slash commands received from Discord servers, including getOptionAsZeroOrPositiveInt(), getOptionAsPositiveInt(), getOptionAsIntInRange(), getOptionAsStringWithLimitedLength(), getOptionAsStringWithValidation(), and getOptionAsIntWithValidation().

Additionally, this commit updates the POM file to add a dependency on the audio-breakout module.

## Test Scope / Change impact:
- Audio's feature

https://user-images.githubusercontent.com/23109467/230770099-b1054266-5673-4799-abce-b2c061106a06.mp4

### CODE REVIEW

File: audio-breakout/src/main/kotlin/tw/waterballsa/utopia/audiobreakout/listeners.kt [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+        // break out rooms
+        for (i in 0 until groupCount) {
+            val roomNumber = i + 1
+            val startIndex = i * roomSize
+            val endIndex = min((i + 1) * roomSize, memberCount)
+            val groupMembers = members.subList(startIndex, endIndex)
+            val groupChannelName = "┗ $roomNumber • \uD83D\uDC65 $roomNamePrefix"
+
+            val groupChannel = if (mainCategory == null) guild.createVoiceChannel(groupChannelName).complete()
```
Comment (frankvicky) [inline audio-breakout/src/main/kotlin/tw/waterballsa/utopia/audiobreakout/listeners.kt:77]:
```
mainCategory
    ?.let { it.createVoiceChannel(groupChannelName).complete() }
    ?: guild.createVoiceChannel(groupChannelName).complete()
```

File: audio-breakout/src/main/kotlin/tw/waterballsa/utopia/audiobreakout/listeners.kt [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+            groupChannel.sendMessage("倒數 $countdownTimeInSeconds 秒鐘，<t:$endTimeInSeconds:t> 集合！").queue()
+        }
+
+        reply("Rooms break-out successfully!").complete()
+
+        timer.schedule(timerTask {
+            for (groupChannel in roomChannels) {
+                groupChannel.members.forEach { guild.moveVoiceMember(it, mainChannel).complete() }
+                groupChannel.delete().queue()
+            }
```
Comment (frankvicky) [inline audio-breakout/src/main/kotlin/tw/waterballsa/utopia/audiobreakout/listeners.kt:101]: `roomChannels.forEach { 
    // 這樣可以使用 it 代替
 }`

File: audio-breakout/src/main/kotlin/tw/waterballsa/utopia/audiobreakout/listeners.kt [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+    }
+}
+
+private fun SlashCommandInteractionEvent.validateMemberShouldNotAlreadyBeInRoom(): Boolean {
+    val memberFound = roomIdToMemberIds.values
+            .flatten().find { it == member?.id }
+    if (memberFound != null) {
+        log.info { "[member should not already be in room] {\"memberId\"=\"${member?.id}\"}" }
+    }
```
Comment (frankvicky) [inline audio-breakout/src/main/kotlin/tw/waterballsa/utopia/audiobreakout/listeners.kt:114]:
```
memberFound
    ?.let { log.info { "[member should not already be in room] {\"memberId\"=\"${member?.id}\"}" } }
```

File: discord-impl-jda/src/main/kotlin/tw/waterballsa/utopia/jda/jda.kt [hunk 4]
```diff
@@ -69,16 +71,21 @@ open class JdaConfig {
...
     }
 
     @Suppress("UNCHECKED_CAST")
     inline fun <reified T : GenericEvent> on(noinline listenerDeclaration: T.() -> Unit) {
         val pair = T::class.java to listenerDeclaration
-        declarations[pair.first] = pair.second as GenericEvent.() -> Unit
+        listenerDeclarations[pair.first] = pair.second as GenericEvent.() -> Unit
+    }
+
+    fun command(commandDeclarations: () -> CommandData) {
```
Comment (Wally5077) [inline discord-impl-jda/src/main/kotlin/tw/waterballsa/utopia/jda/jda.kt:87]: commandDeclaration

**Review by Wally5077 — CHANGES_REQUESTED** [review]
Review done.

**Review by frankvicky — APPROVED** [review]

**Pull request merged**